pkg/operator/config/config.go

```diff
@@ -17,17 +17,34 @@ limitations under the License.
 package config
 
 import (
+	"context"
+	"fmt"
 	"os"
 	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	configv1 "github.com/openshift/api/config/v1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	// TODO (alebedev): CPaaS onboarding is ongoing to replace this with Red Hat built image
-	DefaultExternalDNSImage  = "k8s.gcr.io/external-dns/external-dns:v0.8.0"
-	DefaultMetricsAddr       = "127.0.0.1:8080"
-	DefaultOperatorNamespace = "external-dns-operator"
-	DefaultOperandNamespace  = "external-dns"
-	DefaultEnableWebhook     = true
+	DefaultExternalDNSImage        = "k8s.gcr.io/external-dns/external-dns:v0.8.0"
+	DefaultMetricsAddr             = "127.0.0.1:8080"
+	DefaultOperatorNamespace       = "external-dns-operator"
+	DefaultOperandNamespace        = "external-dns"
+	DefaultEnableWebhook           = true
+	DefaultEnablePlatformDetection = true
+	DefaultIsOpenShift             = false
```

Comment:
`DefaultIsOpenShift` can be removed too along with the `operatorconfig.New` function.